src/main/java/net/glease/ggfab/mte/MTE_AdvAssLine.java

```diff
@@ -844,9 +843,7 @@ public CheckRecipeResult checkProcessing() {
                     // of batches we want to run. The latter is done to prevent batch mode from ever going above
                     // BATCH_MODE_DESIRED_TICKS_PER_SLICE ticks per slice (see also where it is defined above).
                     int parallel = Math.min(recipesAvailable, desiredBatches);
-                    // We no longer need to check if we have enough items in the first slot, as this is
-                    // guaranteed by taking the minimum earlier.
-                    if (hasAllFluids(recipe, parallel)) {
+                    if (hasAllFluids(recipe, parallel) && hasAllItems(recipe, parallel)) {
```

Comment:
Does this properly work with wall-shared AALs with shared input buses and hatches? (https://github.com/GTNewHorizons/GigaGramFab/pull/35)

Reply:
Honestly, I don't know. I assume it cannot run this logic for different AALs in multiple threads so presumably it does work since one will have extracted the fluid by the time the other checks. I don't believe this is much different from how other multiblocks check for recipes/items present, but I could be wrong.